src/common/definitions.js

```diff
@@ -125,6 +125,92 @@ const address = (() => {
   };
 })();
 
+const addressBuilder = (useCountryFullName = false) => {
+  const countries = constants.countries.map(object => (useCountryFullName ? object.label : object.value));
+  const countriesWithAnyState = Object.keys(
+    useCountryFullName ? constants.statesWithFullCountryNames : constants.states,
+  ).filter(x => _.includes(countries, x));
+  const countryStateProperties = _.map(
+    useCountryFullName ? constants.statesWithFullCountryNames : constants.states,
+    (value, key) => ({
+      properties: {
+        country: {
+          type: 'string',
+          enum: [key],
+        },
+        state: {
+          type: 'string',
+          enum: value.map(x => x.value),
+        },
+        postalCode: {
+          type: 'string',
+          maxLength: 10,
+        },
+      },
+    }),
+  );
+  countryStateProperties.push({
+    properties: {
+      country: {
+        not: {
+          type: 'string',
+          enum: countriesWithAnyState,
+        },
+      },
+      state: {
+        type: 'string',
+        maxLength: 51,
+      },
+      postalCode: {
+        type: 'string',
+        maxLength: 51,
+      },
+    },
+  });
+
+  return {
+    type: 'object',
+    oneOf: countryStateProperties,
+    properties: {
+      isMilitaryBase: {
+        type: 'boolean',
+        default: false,
+      },
+      country: {
+        type: 'string',
+      },
+      street: {
+        type: 'string',
+        minLength: 1,
+        maxLength: 50,
+      },
+      street2: {
+        type: 'string',
+        minLength: 1,
```

Comment:
Including a `minLength` makes this field required, I believe.

Reply:
Just removed it.